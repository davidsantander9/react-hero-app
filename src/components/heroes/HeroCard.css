body{
    background: #000;
}

h1{
    color: #fff;
}

h4{
    color: #fff;
}
.my-card {
    
    min-width: 260px;
    max-width: 300px;
    max-height: 70vh;
    border-radius: 5px;
    box-shadow: 0px 0px 25px rgba(0, 0, 0, 0.1);
    overflow: hidden;
    position: relative;
    margin: 10px;
    cursor: pointer;
    display: inline-block;
}

.hero-screen{
    background: #212529!important;
    display: flex;
    flex-direction: row;
    justify-content: center;
    background-repeat: no-repeat;
    background: #fff;
    border-radius: 5px;
    padding: 10px;
}
.hero-data{
    color: #fff;
    display: flex;
    text-align: center;
    padding: 10px;
    flex-direction: column;
    justify-content: space-around;
}

.img-hero img{
    max-height: 500px;
    border-radius: 5px;
    filter: brightness(70%);
}
.my-card img { 
    transition: all 0.15s linear;
    position: relative;
    top: -15%;
    left: -25%;
    filter: brightness(70%);
}
 
.my-card .profile-name {
    position: absolute;
    top: 10px;
    left: 10px;
    font-size: 25px;
    font-weight: bold;
    color: #FFF;
    padding: 15px 20px;
    transition: all 0.15s linear;
}
 
.my-card .profile-position {
    position: absolute;
    color: rgba(255, 255, 255, 0.596);
    left: 30px;
    top: 60px;
    transition: all 0.15s linear;
}
 
.my-card .profile-overview {
    position: absolute;
    bottom: 0px;
    left: 0px;
    right: 0px;
    background: linear-gradient(0deg, rgba(0, 0, 0, 0.4) 50%, rgba(255, 255, 0, 0));
    color: #FFF;
    padding: 50px 0px 40px 0px;
    transition: all 0.15s linear;
}
 
.my-card .profile-overview h3 {
    font-weight: bold;
    font-size: large;
    padding-left: 25px;
}
 
.my-card .profile-overview p {
    color: rgba(255, 255, 255, 0.7);
    padding-left: 25px;
    padding-right: 25px;
}
 
.my-card:hover .profile-name {
    padding-left: 25px;
}
 
.my-card:hover .profile-position {
    left: 40px;
}
 
.my-card:hover .profile-overview {
    padding-bottom: 25px;
}
 
.my-card img:hover {
    transform: scale(1.01);
}