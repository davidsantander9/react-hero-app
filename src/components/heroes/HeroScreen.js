import React, { useMemo } from 'react';
import { Redirect, useParams } from 'react-router-dom';
import { getHeroById } from '../../selectors/getHeroById';
import Color, { Palette } from "color-thief-react";

export const HeroScreen = ({ history }) => {

    const { heroId } = useParams();

    const hero = useMemo(() => getHeroById( heroId ), heroId);


    if( !hero ){
        return <Redirect to="/" />
    }
    const {
        id,
        superhero,
        alter_ego,
        first_appearance,
        characters,
    } = hero

    const path = `../assets/heroes/${id}.jpg`;

    const handleReturn = () => {
        if( history.length <= 2){
            history.push('/');
        }else{
            history.goBack();
        }
    }

    return (

        <Color src={path} crossOrigin="anonymous" format="hex">
                {({ data, loading }) => {
                if (loading) return (<p>No image</p>);
                return (

                    <div className='hero-screen animate__animated animate__flipInX' 
                        style={{
                            backgroundColor: data ,
                        }}
                    >
                        
                        <div className='img-hero'>
                            <img src={path} className="img img-responsive" alt={superhero}/>
                        </div>
                        <div className='hero-data'>
                            <h1>Superhero: {superhero}</h1>
                            <h2>Alter ego: {alter_ego}</h2>
                            <p>First appearance: {first_appearance}</p>
                            <p>Characters: {characters}</p>
                            <button 
                                className='btn btn-primary' 
                                onClick={ handleReturn }
                            >
                                Regresar
                            </button>
                        </div>
                    </div>
                );
                }}
        </Color>
        
    )
};
