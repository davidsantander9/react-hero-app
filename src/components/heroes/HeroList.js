import React, { useMemo } from 'react'
import { getHeroesByPublisher } from '../../selectors/getHeroByPublisher'
import { HeroCard } from './HeroCard';
import "./HeroCard.css";

export const HeroList = ({ publisher }) => {

    const heroes = useMemo(() => getHeroesByPublisher(publisher), [publisher])

    return (
        <div className='card-box' >
            {
                heroes.map( hero => (
                    <HeroCard key={hero.id} {...hero}/>
                ))
            }
        </div>
    )
}
