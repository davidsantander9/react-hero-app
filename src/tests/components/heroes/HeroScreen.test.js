import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter, Route } from 'react-router';

import { HeroScreen } from '../../../components/heroes/HeroScreen';


describe('test on <HeroScreen/>', () => {
    
    const historyMock = {
        length: 10,
        push: jest.fn(),
        goBack: jest.fn() 
    }

    
    test('should show <redirect/> is there wasnt args in URL', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/hero']}>
                <HeroScreen history = { historyMock }/>
            </MemoryRouter>
        );
        expect( wrapper ).toMatchSnapshot();
    });


    // test('should show a hero if param exist and found it', () => {
    //     const wrapper = mount(
    //         <MemoryRouter initialEntries={['/hero/marvel-spider']}>
    //             <Route path="/hero/:heroId" component={ HeroScreen }/>
    //         </MemoryRouter> 
    //     );

    //     console.log(wrapper);
    //     expect( wrapper.find('.row').exists()).toBe(true);
    // });


    // test('should return to previous page', () => {

    //     const history = {
    //         length: 1,
    //         push: jest.fn(),
    //         goBack: jest.fn() 
    //     }

    //     const wrapper = mount(
    //         <MemoryRouter initialEntries={['/hero/dc-robin']}>
    //             <Route 
    //             path="/hero/:heroId" 
    //             component={ () =>  <HeroScreen history={ history }/> }
    //         />
    //         </MemoryRouter>
    //     );

    //     wrapper.find('button').prop('onClick')();

    //     expect( history.push ).toHaveBeenCalledWith('/');
    //     expect( history.goBack ).not.toHaveBeenCalled();
    // });
    

})
