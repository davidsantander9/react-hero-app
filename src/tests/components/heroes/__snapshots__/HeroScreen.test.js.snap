// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <HeroScreen/> should show <redirect/> is there wasnt args in URL 1`] = `""`;
